proposals/lip-0050.md

```diff
@@ -248,8 +408,8 @@ Then, do the following:
 
 - Check if the `address` properties of the entries in `genesisBlockAssetObject.accounts` are pairwise distinct. If not, reject the block.
 - Check if the sum of the `balance` properties of the entries in `genesisBlockAssetObject.accounts` is less than 2<sup>64</sup>. If not, reject the block.
+- Check if the sum of the `balance` properties of the entries in `genesisBlockAssetObject.accounts` equals `token.getLockedAmount(ADDRESS_LEGACY_RESERVE, MODULE_ID_LEGACY, TOKEN_ID_LSK_MAINCHAIN)`.
```

Comment:
If we want to do it like this, we need to update LIP 63 to create this account with the correct balance. Alternatively, we could instead mint the amount of tokens here and lock it (in the genesis state finalization stage). 

The advantage of the current approach is that the genesis block asset of the Token module contains the correct genesis supply of `LSK`. The advantage of the alternative approach is that it would make the genesis state of the Token module and Legacy module more independent (the genesis asset of the Token module does not need to be initialized based on the state of the legacy module) and it also would not require an update of LIP 63.

@sergeyshemyakov @MaximeGagnebin What do you guys think?

Reply:
Personally, I prefer the current writing.

Reply:
As discussed in our stand up, we would leave the writing as is. The migration LIP should then be updated to create the account with `address = ADDRESS_LEGACY_RESERVE` and `locked balance = sum of all reclaim balances` in the genesis token asset.